### PR TITLE
Fix/seats-cancel-links

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Add test to make sure seat cancellation doesn't cancel whole booking for everyone
- Fixes some error while generating cancel link on booking success page and email.

Fixes #

**Environment**: Staging(main branch)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Setup an event type with advanced options with seats
- [ ] Do several bookings on that and validate both emails cancel link and booking success page cancel action

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -1429,6 +1429,7 @@ async function handler(
        * deep cloning evt to avoid this
        */
       const copyEvent = cloneDeep(evt);
+      copyEvent.uid = booking.uid;
```
Comment (alannnc) [inline packages/features/bookings/lib/handleNewBooking.ts:1432]: uid wasn't passed down to emails so cancel link arrived with wrong booking.uid

File: packages/lib/CalEventParser.ts [hunk 2]
```diff
@@ -150,7 +150,7 @@ export const getUid = (calEvent: CalendarEvent): string => {
 };
 
 const getSeatReferenceId = (calEvent: CalendarEvent): string => {
-  return calEvent.attendeeSeatId ? `seatReferenceUid=${calEvent.attendeeSeatId}` : "";
+  return calEvent.attendeeSeatId ? calEvent.attendeeSeatId : "";
```
Comment (alannnc) [inline packages/lib/CalEventParser.ts:153]: This was returning a query param for both scenarios and one of them wasn't required so the result was a malformed url.

**Comment by CarinaWolli** [comment]
When clicking on the cancel link from the email I can see all seats booked, could we fix that in this PR too? 
Screenshot 2023-04-20 at 11 43 03

**Review by emrysal — APPROVED** [review]

File: packages/lib/CalEventParser.ts [hunk 3]
```diff
@@ -161,12 +161,12 @@ ${WEBAPP_URL + "/booking/" + getUid(calEvent) + "?changes=true"}
...
-    `/booking/${getUid(
-      calEvent
-    )}?cancel=true&allRemainingBookings=${!!calEvent.recurringEvent}&${getSeatReferenceId}`
-  );
+  const cancelLink = new URL(WEBAPP_URL + `/booking/${getUid(calEvent)}`);
+  cancelLink.searchParams.append("cancel", "true");
+  cancelLink.searchParams.append("allRemainingBookings", String(!!calEvent.recurringEvent));
+  const seatReferenceUid = getSeatReferenceId(calEvent);
+  if (seatReferenceUid) cancelLink.searchParams.append("seatReferenceUid", seatReferenceUid);
+  return cancelLink.toString();
```
Comment (alannnc) [inline packages/lib/CalEventParser.ts:169]: Looks good!

**Review by zomars — APPROVED** [review]

**Review by emrysal — APPROVED** [review]

File: apps/web/playwright/booking-seats.e2e.ts [hunk 4]
```diff
@@ -313,5 +313,145 @@ test.describe("Booking with Seats", () => {
...
+      });
+
+      // Go to cancel page and see that attendees are listed and myself as I'm owner of the booking
+      await page.goto(`/booking/${booking.uid}?cancel=true&allRemainingBookings=false`);
+
+      const foundFirstAttendeeAsOwner = await page.locator('p[data-testid="attendee-first+seats@cal.com"]');
+      await expect(foundFirstAttendeeAsOwner).toHaveCount(1);
+      const foundSecondAttendeeAsOwner = await page.locator('p[data-testid="attendee-second+seats@cal.com"]');
+      await expect(foundSecondAttendeeAsOwner).toHaveCount(1);
+      await page.pause();
```
Comment (zomars) [inline apps/web/playwright/booking-seats.e2e.ts:418]: This is causing e2e to stall on main 😓

**Pull request merged**